main.py

```diff
@@ -0,0 +1,108 @@
+    average = sum(closing_prices) / len(closing_prices)
+    print(f"{ticker} Average Closing Price: {average:.2f}")
+    
+
+# --------- REAL-TIME PRICE -----------
+# Function to retrieve real-time price
+def get_realtime_price(api):
+    tickers = ["TSLA", "AAPL", "MSFT", "GOOG", "NKE"]
+    response = {}
+    for ticker in tickers:
+        url = f"https://api.twelvedata.com/price?symbol={ticker}&apikey={api}"
+        response[ticker] = requests.get(url).json()
+    return response
+
+# Store real-time price data
+data_realtime = get_realtime_price(config.api_key2)
+
+
+# --------- MARKET OPEN PRICE -----------
+# Function to get the market open, to compare real-time price to
+def get_market_open(api, today):
+    tickers = ["TSLA", "AAPL", "MSFT", "GOOG", "NKE"]
+    response = {}
+    for ticker in tickers:
+        url = f"https://api.twelvedata.com/time_series?symbol={ticker}&interval=1day&outputsize=1&apikey={api}"
+        response[ticker] = requests.get(url).json()
+    return response
+
+# Store market open price data
+data_marketopen = get_market_open(config.api_key3, today)
+
+# print for visualisation purposes:
+for ticker, stock_data in data_marketopen.items():
+    open_prices = [float(day['open']) for day in stock_data['values']]
+    print(f"{ticker} Market Open: {open_prices}")
+
+# --------- TRIGGER WEBHOOK IF CONDITION IS MET -----------
+# Function to trigger IFTTT event
+def trigger(event_name):
+    url = f'https://maker.ifttt.com/trigger/{event_name}/with/key/{config.ifttt_key}'
+    response = requests.post(url)
+    if response.status_code == 200:
+        print(f"Triggered the IFTTT applet for {event_name} successfully.")
+    else:
+        print(f"Failed to trigger the IFTTT applet for {event_name}.")
+
+# Map of tickers to event names
+ticker_to_event = {
+    "TSLA": "TSLA_Price_Drop",
+    "AAPL": "AAPL_Price_Drop",
+    "MSFT": "MSFT_Price_Drop",
+    "GOOG": "GOOG_Price_Drop",
+    "NKE": "NKE_Price_Drop"
+}
+
+# Converting £0.25 to USD with current exchange rate 
+# and assigning it a variable to calculate the variance between real time and market open
+def currency_conversion(api):
+    response = {}
+    url = f"https://api.twelvedata.com/currency_conversion?symbol=GBP/USD&amount=0.25&apikey={api}"
+    response = requests.get(url).json()
+    return response
+
+pence_to_cents = currency_conversion(config.api_key1)
+pence_to_cents = pence_to_cents['amount']
+print(f"Current USD value of £0.25: {pence_to_cents:.2f}")
+
+# Check conditions and trigger the corresponding event
+for ticker, stock_data in data_realtime.items():
+    current_price = float(stock_data['price'])
+    event_name = ticker_to_event.get(ticker)
+    market_open_price = float(data_marketopen[ticker]['values'][0]['open'])
+    print(f"{ticker} Current Price: {current_price:.2f}")
+
+    if current_price <= market_open_price - pence_to_cents:
+        trigger(event_name)
+    elif current_price < average:
+        trigger(event_name)
```

Comment:
You need an if name == main block at the end. Any code inside this block is what gets executed whenever you run this file in the terminal